### ISSUE TITLE
Have a `renew` endpoint that verifies if the student account is still valid

### ISSUE THREAD

**Issue by ghostwriternr** [metakgp/leprechaun#11]
# Idea
This could be fairly trivial, but it would be nice if we have a `renew` endpoint that checks if the ERP account is still active. ERP doesn't even show the Security Question box when entering an old roll number (ex, 11CS30040).

## Why?
Currently, Leprechaun doesn't an API consumer know if a particular roll number is still a student. But this would be pretty useful if we integrate Leprechaun with MFTP, where we might want to remove past students automatically.

**Comment by icyflame** [comment]
So, we have had this conversation before. I am not sure where it was, but I am pretty sure it was part of a Demo day discussion. The solution that we came up with was:

1. we would return the timestamp for when the user authenticated with us last (TODO) 
1. If the application that is consuming this data needs a more recent confirmation, then it can ask the user to re-authenticate

**Note:** Re-authentication is exactly same as the original authentication flow. If they do the whole thing again, their returned date and time will be updated.

This has the advantage of Leprechaun being simultaneously useful for applications that serve current KGP students and one-time KGP students without affecting our handling of their data. Let me know if something seems off about this. I will be making the change to return the time they authenticated with Leprechaun in a bit.

**Comment by icyflame** [comment]
So, we have had this conversation before. I am not sure where it was, but I am pretty sure it was part of a Demo day discussion. The solution that we came up with was:

1. we would return the timestamp for when the user authenticated with us last (TODO) 
1. If the application that is consuming this data needs a more recent confirmation, then it can ask the user to re-authenticate

**Note:** Re-authentication is exactly same as the original authentication flow. If they do the whole thing again, their returned date and time will be updated.

This has the advantage of make Leprechaun simultaneously useful for applications that serve current KGP students only and one-time KGP students without affecting either. Let me know if something seems off about this. I will be making the change to return the time they authenticated with Leprechaun in a bit.

**Comment by ghostwriternr** [comment]
Oh, but do we really need the timestamp though? Because once we connect someone's roll number with their email address and say an API consumer wants to know if the person is still a student, all we need to do is to verify they are still a student (and not make them re-authenticate, because it's unnecessary?). All we'd need to do is go to the ERP, verify if the associated roll number still returns some security question. If it doesn't, well they are not a student anymore. I don't think there are any flaws with doing it this way.

I'm not against timestamps, but I don't think it serves any purpose for the sake of this issue.

**Comment by icyflame** [comment]
Yeah, but notice that there are more corner cases. (The case when the ERP is down and the application is checking if someone is a current student is one that comes to mind)

Of course, as you said, this can be done by the application. So, from Leprechaun's side, there are no changes required. The application can institute any check of their own (like a simple Regex check and matching the current time and expected time of graduation, check for ERP security questions, make them re-auth, etc) Not our concern. :slightly_smiling_face: 

I wanted to include the time stamp of authentication anyway, because without it, the information is very plain. Just a roll number and an email ID connection without knowing how stale the information is is a bit much for developers. (Say they are trying to debug why 10/100 invitees of a service are not replying, with a date, they can evaluate how old the information is and make some assumption from there at least) So, I have included the time stamp anyway.

I think we can close this issue?

**TL; DR** Renewal is same as re-authenticating with the same roll number and email ID.
